src/main/webapp/app/assessment/athena.service.ts

```diff
@@ -14,42 +18,119 @@ export class AthenaService {
+            }),
+        );
     }
 
     /**
-     * Get feedback suggestions for the given submission from Athena - for text exercises
-     * Currently, this is separate for programming and text exercises (will be changed)
+     * Find a grading instruction by id in the given exercise
+     */
+    private findGradingInstruction(exercise: Exercise, id: number): any | undefined {
+        for (const criterium of exercise.gradingCriteria ?? []) {
+            for (const instruction of criterium.structuredGradingInstructions) {
+                if (instruction.id == id) {
+                    return instruction;
+                }
+            }
+        }
+        return undefined;
+    }
+
+    /**
+     * Get feedback suggestions for the given text submission from Athena
+     *
+     * @param exercise
+     * @param submission  the submission
+     * @return observable that emits the referenced feedback suggestions as TextBlockRef objects
+     * with TextBlocks and the unreferenced feedback suggestions as Feedback objects
+     * with the "FeedbackSuggestion:" prefix
+     */
+    public getTextFeedbackSuggestions(exercise: Exercise, submission: TextSubmission): Observable<(TextBlockRef | Feedback)[]> {
+        return this.getFeedbackSuggestions<TextFeedbackSuggestion>(exercise, submission.id!).pipe(
+            map((suggestions) => {
+                // Convert referenced feedback suggestions to TextBlockRefs for easier handling in the components
+                return suggestions.map((suggestion) => {
+                    const feedback = new Feedback();
+                    feedback.credits = suggestion.credits;
+                    // Text feedback suggestions are automatically accepted, so we can set the text directly:
+                    feedback.text = FEEDBACK_SUGGESTION_ACCEPTED_IDENTIFIER + suggestion.title;
+                    feedback.detailText = suggestion.description;
+                    // Load grading instruction from exercise, if available
+                    if (suggestion.structuredGradingInstructionId != undefined) {
+                        feedback.gradingInstruction = this.findGradingInstruction(exercise, suggestion.structuredGradingInstructionId);
+                    }
+                    if (suggestion.indexStart == null) {
+                        // Unreferenced feedback, return Feedback object
+                        feedback.type = FeedbackType.MANUAL_UNREFERENCED;
+                        return feedback;
+                    }
+                    // Referenced feedback, convert to TextBlockRef
+                    feedback.type = FeedbackType.MANUAL;
+                    const textBlock = new TextBlock();
+                    textBlock.startIndex = suggestion.indexStart;
+                    textBlock.endIndex = suggestion.indexEnd;
+                    textBlock.setTextFromSubmission(submission);
+                    feedback.reference = textBlock.id;
```

Comment:
I think a `textBlock.computeId()` is missing or how is it set?

Reply:
No longer needed because of #7134